src/etc/inc/system.inc

```diff
@@ -1504,6 +1525,17 @@ EOD;
 		$nginx_config .= "\t\tadd_header X-Content-Type-Options nosniff;\n";
 		$nginx_config .= "\t\tssl_session_tickets off;\n";
 		$nginx_config .= "\t\tssl_dhparam /etc/dh-parameters.4096;\n";
+		if (($config['system']['webgui']['ocsp-staple'] == true) ||
+		    cert_get_ocspstaple(lookup_cert($config['system']['webgui']['ssl-certref'])) {
+			$nginx_config .= "\t\tssl_stapling on;\n";
+			$nginx_config .= "\t\tssl_stapling_verify on;\n";
+			$nginx_config .= "\t\tresolver ";
+			foreach(get_dns_nameservers() as $dns_ns) {
```

Comment:
This could be optimized further, instead of doing a foreach, replace that block with:

    $nginx_config .= implode(" ", get_dns_nameservers()) . " valid=300s;\n";